### PR TITLE
feat(inference/tts): port aligned transcript / output_timestamps from Python (#5534)

### PR DESCRIPTION
**Pull request by toubatbrian**
## Summary

Automated port of [livekit/agents#5534](https://github.com/livekit/agents/pull/5534) (*"feat(tts): add support for timestamps in Inference"*) into `agents-js`.

The upstream Python PR taught the Inference TTS client to surface the LiveKit Inference gateway's new `output_timestamps` WebSocket event as word- / character-level `TimedString`s, and to advertise the `aligned_transcript` capability automatically whenever a provider-specific opt-in flag is present in `extra_kwargs`. This PR brings the same behaviour to the Node.js client.

Triggered by the automated Claude Code routine (experimental).

## Scope

This change is a plugin-layer improvement — all edits are confined to `agents/src/inference/` and its tests. No public API surface changes outside of the inference TTS namespace.

### 1. `hasAlignedTranscript(model, modelOptions)` helper

Direct analogue of the Python `_has_aligned_transcript`. Inspects the provider prefix and returns `true` if the provider-specific alignment opt-in is present in `modelOptions`:

| Provider | `modelOptions` flag that enables alignment |
| --- | --- |
| `cartesia/*` | `add_timestamps: true` |
| `elevenlabs/*` | `sync_alignment: true` |
| `inworld/*` | `timestamp_type: 'WORD' \| 'CHARACTER'` |

Any other provider (e.g. `deepgram`, `rime`) returns `false`, matching Python.

### 2. Dynamic `TTSCapabilities.alignedTranscript`

- Computed at construction from the incoming `model` + `modelOptions` and forwarded to `super(..., { streaming: true, alignedTranscript })`.
- Re-computed inside `updateOptions` when either `model` or `modelOptions` changes, mirroring Python's `self._capabilities.aligned_transcript = _has_aligned_transcript(...)`.
- Exposed via an overridden `get capabilities()` on the inference TTS subclass.

### 3. Runtime `modelOptions` updates

`TTS.updateOptions` / `SynthesizeStream.updateOptions` now accept `modelOptions` in addition to `model` / `voice` / `language`. The new options are merged shallowly into the existing `modelOptions`, matching Python's `self._opts.extra_kwargs.update(extra_kwargs)`.

### 4. `output_timestamps` WebSocket event

- Added `ttsOutputTimestampsEventSchema` to `agents/src/inference/api_protos.ts` as a new branch of the `ttsServerEventSchema` discriminated union. The schema supports both `words: [{word, start, end}]` and `chars: [{char, start, end}]` payloads.
- `SynthesizeStream.run` now buffers decoded timings as `TimedString[]` (`pendingTimedTranscripts`) and attaches them to the next audio frame via `SynthesizedAudio.timedTranscripts`. Matches the Cartesia plugin's pattern for word-level timing and the downstream `TranscriptionSynchronizer` consumer.

## Implementation notes where JS differs from Python

Code-level parity is near-1:1, modulo the following language-level adjustments:

1. **Capability mutation.** Python mutates `self._capabilities.aligned_transcript` in place on the dataclass held on the base `TTS`. The JS base class stores capabilities in a hard-private `#capabilities` field accessed through a getter, so mutation from a subclass isn't possible. Instead, the inference TTS subclass tracks its own `#alignedTranscript` boolean and **overrides** `get capabilities()` to return the live value. Externally observable behaviour is identical — `tts.capabilities.alignedTranscript` reflects the latest `modelOptions`.
2. **`push_timed_transcript` → `SynthesizedAudio.timedTranscripts`.** Python exposes `output_emitter.push_timed_transcript(ts)`, which streams timed strings alongside audio. The JS agents framework has no such abstraction; the established pattern (see `plugins/cartesia/src/tts.ts`) is to buffer `TimedString[]` and attach them to the next `SynthesizedAudio` packet via the optional `timedTranscripts` field. This is what `TranscriptionSynchronizer` already consumes.
3. **`update_options` signature.** The Python `update_options` had always accepted `extra_kwargs`; JS `updateOptions` previously did not. To preserve Python semantics (recomputing alignment when options change at runtime), the JS `updateOptions` signature now also accepts a partial `modelOptions` patch — shallow-merged into the stored options. This is a strict additive change to the type; existing callers compile unchanged.
4. **Text of the `TimedString`.** Python passes the raw `word_info["word"]` / `char_info["char"]` with no trailing whitespace. The JS port does the same — unlike the Cartesia plugin port, no `+ ' '` is appended. If padding is desired in the future we can align both ports in a follow-up.
5. **Type guardrails.** `modelOptions` is carried as the generic `TTSOptions<TModel>` type on the way in, but `hasAlignedTranscript` intentionally accepts `Record<string, unknown>` to avoid having to widen every provider interface just to read three opt-in fields.

## Files changed

- `agents/src/inference/api_protos.ts` — new `ttsWordTimestampSchema` / `ttsCharTimestampSchema` / `ttsOutputTimestampsEventSchema` and their inferred types, added to the server-event union.
- `agents/src/inference/tts.ts` — new `hasAlignedTranscript` helper, dynamic capability wiring in constructor + `updateOptions`, `output_timestamps` handling in `SynthesizeStream.run`, `TimedString` buffer attached to the next frame.
- `agents/src/inference/tts.test.ts` — new `describe` blocks for `hasAlignedTranscript` (provider matrix + edge cases) and `TTS alignedTranscript capability` (constructor-time computation and `updateOptions` recomputation).
- `.changeset/inference-tts-aligned-timestamps.md` — minor changeset entry.

## Cross-reference comments

Every ported section carries an inline `// Ref: python livekit-agents/livekit/agents/inference/tts.py - <lines>` comment pointing back at the corresponding lines in the upstream diff, per the repo's porting convention.

## Test plan

- [x] `pnpm build:agents` succeeds (tsup + `tsc --declaration`).
- [x] `pnpm exec vitest run src/inference/tts.test.ts src/inference/api_protos.test.ts` — 51 tests green.
- [x] `pnpm exec eslint src/inference/tts.ts src/inference/tts.test.ts src/inference/api_protos.ts` — no errors introduced by this PR (4 pre-existing tsdoc warnings on `Range >0.5, <=1.5.` JSDoc in `InworldOptions` / `RimeOptions`, unchanged).
- [x] `pnpm format:check` clean.
- [ ] Manual smoke test against the LiveKit Inference gateway with an ElevenLabs model and `modelOptions: { sync_alignment: true }` — confirm `output_timestamps` frames flow through to the `TranscriptionSynchronizer`.
- [ ] Manual smoke test with Cartesia `add_timestamps: true` and Inworld `timestamp_type: 'WORD'`.

---

This is an automated port from [livekit/agents#5534](https://github.com/livekit/agents/pull/5534) by the Claude Code automation routine (experimental).

cc @toubatbrian @livekit/agent-devs

---
_Generated by [Claude Code](https://claude.ai/code/session_01ToP9m3meuAEg56JPgUdQKU)_

### CODE REVIEW

File: agents/src/inference/tts.ts [hunk 1]
```diff
@@ -318,17 +358,42 @@ export class TTS<TModel extends TTSModels> extends BaseTTS {
...
+
+    // Ref: python livekit-agents/livekit/agents/inference/tts.py - 540-542 lines
+    this.#alignedTranscript = hasAlignedTranscript(
+      this.opts.model,
+      this.opts.modelOptions as Record<string, unknown>,
+    );
+
     for (const stream of this.streams) {
       stream.updateOptions(this.opts);
     }
```
[the remark this answers is not in the document]
Comment (toubatbrian) [reply agents/src/inference/tts.ts:408]: Good catch — thank you. You're right that `session.create` is only sent on the first open, so a pooled socket would continue running with the old session while `capabilities.alignedTranscript` reports the post-update value. In TTS-aligned-transcript flows that would silently drop the timed-text stream.

Fixed in 4cedebc: `updateOptions` now calls `this.pool.invalidate()` whenever any session-affecting option (`model`, `voice`, `language`, or `modelOptions`) changes, so the next `stream()` opens a fresh socket with the up-to-date `session.create` payload. The existing in-flight streams still get their `opts` patched for the non-session aspects they use, but new connections will carry the new options. Added a unit test that spies on `pool.invalidate` to pin the behaviour.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by lukasIO — APPROVED** [review]
lgtm!
